hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -778,6 +837,49 @@ withRetries :: ShrinkRetries -> Property -> Property
 withRetries n =
   mapConfig $ \config -> config { propertyShrinkRetries = n }
 
+-- | Add a classifier to the test if the predicate is true
+--
+-- @
+--    prop_with_classifier :: Property
+--    prop_with_classifier = property $ do
+--      xs <- forAll $ Gen.list (Range.linear 0 100) Gen.alpha
+--      for_ xs $ \x ->
+--        classify (x == 0) "newborns" $ do
+--        classify (x > 0 && x < 13) "children" $
+--        classify (x > 12 && x < 20) "teens" $
+--          success
+-- @
+classify :: Bool -> String -> PropertyT IO () -> PropertyT IO ()
+classify = cover 0
+
+cover :: Double -> Bool -> String -> PropertyT IO () -> PropertyT IO ()
```

Comment:
Similar to `annotate` in hedgehog, I think that
```haskell
cover :: (MonadTest m, HasCallStack) => Double -> Bool -> String -> m ()
```
would be a better type signature, and play nicer with how hedgehog tests are usually written.